### DISCUSSION TITLE
Milestone plan

### DISCUSSION THREAD

**Discussion by mateyab** [SENG-350-2024-fall/Team12#22] [Milestone 3]
Design Patterns:
- Josephine and Nahir
Availability Tactics:
- Ryan and Mateya
Meet next lab and discuss the choices made
Everyone updates their own UML diagram
Split up coding to match design patterns and availability tactics
Implement Nurse portal, triage page, and triage form.

**Comment by nahirv** [comment]
Design Patterns: Decorator, Factory method, Singleton, Strategy, Chain of responsibility

**Comment by mateyab** [comment]
availability tactics

1. transactions - prevents faults
2. ping/echo - detect faults (make sure server is running and reachable)
3. heartbeat - detect faults ( set up a script that will call ping over and over)
4. exception handling - recover from faults ( passwords and any routing / bad insertion to database)
5. rollback - recover from faults (rollback failed aborted transactions and revert to previous state when attempting to insert to databases)

**Comment by mateyab** [comment]
Coding plan:

Tasks ->

**Mateya**
update the home page for many users 
update nurse log-in functionality
this includes creating mock data in the database for pre-established accounts. 
incorporate design patterns 

**Josephine**
Create patient form. 

Implementing tactics:
- transactions and rollback for patient login (Ryan) 
- ping echo + heart beat (Nahir) 

update incremental stuff - Ryan 

update system description - Nahir 
if you have time create a nurse page:
the nurse page will come from  a separate log in page must have the ability to view all the submitted triage forms
and must have the ability to add to the queue ( queue isn't  implemented yet so just a button or something that leads to nothing) 

Everyone update the uml diagrams. Decorator pattern will be very important here
